language-plutus-core/src/Language/PlutusCore/Renamer.hs

```diff
@@ -80,9 +80,9 @@ annotateT (Constant x c) =
 annotateT (TyInst x t ty) =
     TyInst x <$> annotateT t <*> annotateTy ty
 annotateT (Wrap x (TyName (Name x' b u@(Unique i))) ty t) = do
-    aty <- annotateTy ty
     let k = Type x'
     insertKind i k
+    aty <- annotateTy ty
```

Comment:
What does this do?

Reply:
Read the commit message :) this is an unrelated bug which we didn't notice because there are no tests for wrap/unwrap. I'm going to pull them out in a bit.